src/qt/notificator.cpp

```diff
@@ -31,6 +31,18 @@
 #ifdef USE_DBUS
 // https://wiki.ubuntu.com/NotificationDevelopmentGuidelines recommends at least 128
 const int FREEDESKTOP_NOTIFICATION_ICON_SIZE = 128;
+
+void DBusInitThread::run() {
+    auto interface = new QDBusInterface("org.freedesktop.Notifications", "/org/freedesktop/Notifications", "org.freedesktop.Notifications");
+    if (!interface->isValid()) {
+        delete interface;
+        return;
+    }
+    interface->moveToThread(m_notificator.thread());
+    m_notificator.interface = interface;
+    m_notificator.mode = Notificator::Freedesktop;
+}
```

Comment:
Emit [finished](https://doc.qt.io/qt-5/qthread.html#finished) here and clean up instead of keeping the thread for the `Notificator` lifecycle? `DBusInitThread` name implies it's disposable as soon as init is done.

Reply:
Isn't that supposed to be an automatic part of Qt? The docs say the user can't emit it...

Reply:
You're right. I think it used to be public signal but seems that is no longer the case.